### PR TITLE
Neo4j database plugin

### PR DESCRIPTION
**Pull request by eh-steve**
Addresses https://github.com/hashicorp/vault/issues/4046

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -120,7 +120,7 @@ jobs:
     environment:
     - CIRCLECI_CLI_VERSION: 0.1.5546
     - GO_TAGS: ''
-    - GO_VERSION: 1.17.11
+    - GO_VERSION: 1.18.3
```
Comment (cipherboy) [inline .circleci/config.yml:123]: Out of curiosity, is there a change in this PR that requires the updated Go version?

Updating Go versions is a little more involved than this (see some past PRs for example) and is usually something we do closer to release time.

**Review by swenson — COMMENTED** [review]
Why not just put this in a separate repository and build as an external plug-in?

**Comment by eh-steve** [comment]
> Why not just put this in a separate repository and build as an external plug-in?

I can do, I just got the impression from the open issue that it's something that might be wanted built in.

Happy to move it elsewhere and close this PR if you prefer?

**Comment by swenson** [comment]
@eh-steve I think at this point we'd probably just want to have it as a separate plugin, just to avoid plugin sprawl in the main `vault` repo.

Thanks!

File: .circleci/config.yml (hunk 1 above)
Comment (swenson) [reply .circleci/config.yml:123]: This would cause the tests to fail due to the SHA-1 deprecation, so even more reason to move to a separate repo. :)

**Comment by eh-steve** [comment]
Moved to https://github.com/vivacitylabs/neo4j-database-plugin

**Pull request closed without merging**